util/azure/utils.go

```diff
@@ -153,6 +180,26 @@ func NewOperationsMap() OperationsMap {
 	return make(map[string]ResourceAndVerbMap)
 }
 
+func GetOperationsMap() OperationsMap {
+	return operationsMap
+}
+
+func LockOperationsMap() {
+	operationsMapLock.Lock()
+}
+
+func UnlockOperationsMap() {
+	operationsMapLock.Unlock()
+}
+
+func RLockOperationsMap() {
+	operationsMapLock.RLock()
+}
+
+func RUnlockOperationsMap() {
+	operationsMapLock.RUnlock()
+}
```

Comment:
We are making its caller responsibility to maintain synchronization while accessing data. Instead let's explore if we can return copy with read lock to make read operation thread safe.

Reply:
as discussed offline, instead of exposing the lock , in checkaccesshelper we will deepcopy the map and use that version for the calls